moz_kinto_publisher/main.py

```diff
@@ -646,85 +652,101 @@ def clear_crlite_stashes(*, rw_client, noop):
 
 
 def publish_crlite_record(
-    *, path, filename, timestamp, rw_client, incremental, noop, previous_id=None
+    *,
+    attributes,
+    attachment_path,
+    attachment_name,
+    rw_client,
+    noop,
 ):
-    record_type = "diff" if incremental else "full"
-    record_time = timestamp.isoformat(timespec="seconds")
-    record_epoch_time_ms = math.floor(timestamp.timestamp() * 1000)
-    identifier = f"{record_time}Z-{record_type}"
-
-    attributes = {
-        "details": {"name": identifier},
-        "incremental": incremental,
-        "effectiveTimestamp": record_epoch_time_ms,
-    }
-    perms = {"read": ["system.Everyone"]}
-    if incremental:
-        assert previous_id, "Incremental records must have a previous record ID"
-        attributes["parent"] = previous_id
-
-    log.info(
-        f"Publishing {path} {timestamp} incremental={incremental} (previous={previous_id})"
-    )
     if noop:
         log.info("NoOp mode enabled")
+        return attributes["details"]["name"]
 
-    if not noop:
-        record = rw_client.create_record(
+    record = rw_client.create_record(
+        collection=settings.KINTO_CRLITE_COLLECTION,
+        data=attributes,
+        permissions={"read": ["system.Everyone"]},
+    )
+    recordid = record["data"]["id"]
+
+    try:
+        rw_client.attach_file(
             collection=settings.KINTO_CRLITE_COLLECTION,
-            data=attributes,
-            permissions=perms,
+            fileName=attachment_name,
+            filePath=attachment_path,
+            recordId=recordid,
+        )
+    except KintoException as ke:
+        log.error(
+            f"Failed to upload attachment. Removing stale MLBF record {recordid}: {ke}"
```

Comment:
This is also used for stashes, right? Let's replace `MLBF` with `CRLite` or something more generic.

Reply:
Done